spacy/training/loop.py

```diff
@@ -305,3 +311,20 @@ def before_to_disk(nlp: Language) -> Language:
         return modified_nlp
 
     return before_to_disk
+
+
+def clean_output_dir(path: Union[str, Path], *, msg: Printer, stdout: IO) -> None:
+    """Remove an existing output directory. Typically used to ensure that that
+    a directory like model-best and its contents aren't just being overwritten
+    by nlp.to_disk, which could preserve existing subdirectories (e.g.
+    components that don't exist anymore).
+    """
+    if path is not None and path.exists():
+        for subdir in [path / DIR_MODEL_BEST, path / DIR_MODEL_LAST]:
+            if subdir.exists():
+                try:
+                    shutil.rmtree(str(subdir))
+                    logger.debug(f"Removed existing output directory: {subdir}")
+                except Exception as e:
+                    err = Warnings.W087.format(path=subdir, err=e)
+                    stdout.write(msg.warn(err) + "\n")
```

Comment:
This would go to `stderr` rather than stdout I think. But I do think we should just let the error happen.